crypto/asn1/asn_mime.c

```diff
@@ -663,7 +663,10 @@ static STACK_OF(MIME_HEADER) *mime_parse_hdr(BIO *bio)
                     mime_debug("Found End Value\n");
                     *p = 0;
                     mhdr = mime_hdr_new(ntmp, strip_ends(q));
-                    sk_MIME_HEADER_push(headers, mhdr);
+                    if (mhdr == NULL)
+                        goto err;
+                    if (!sk_MIME_HEADER_push(headers, mhdr))
+                        goto err;
```

Comment:
We really should set `mhdr` to NULL here after a successful push. There's a double free waiting to happen if someone adds a new `goto err` call. And the similar call below.